src/qibolab/platforms/platform.py

```diff
@@ -85,3 +115,51 @@ def set_bias(self, qubit, bias):
 
     def get_bias(self, qubit):
         return self.qubits[qubit].flux.bias
+
+
+class AcquisitionType(Enum):
+    """
+    Types of data acquisition from hardware.
+
+    SPECTROSCOPY: Zurich Integration mode for RO frequency sweeps,
+    INTEGRATION: Demodulate and integrate the waveform,
+    RAW: Acquire the waveform as it is,
+    DISCRIMINATION: Demodulate, integrate the waveform and discriminate among states based on the voltages
+
+    """
+
+    RAW = auto()
+    INTEGRATION = auto()
+    DISCRIMINATION = auto()
+
+
+class AveragingMode(Enum):
+    """
+    Types of data averaging from hardware.
+
+    CYLIC: Better averaging for noise,
+    SINGLESHOT: False averaging,
+    [SEQUENTIAL: Worse averaging for noise]
+
+    """
+
+    CYCLIC = auto()
+    SINGLESHOT = auto()
+
+
+@dataclass
+class ExecutionParameters:
+    """Data structure to deal with execution parameters
+
+    :nshots: Number of shots per point on the experiment
+    :relaxation_time: Relaxation time for the qubit
+    :fast_reset: Enable or disable fast reset
+    :acquisition_type: Data acquisition mode
+    :averaging_mode: Data averaging mode
+    """
+
+    nshots: Optional[np.uint32] = 1024
+    relaxation_time: Optional[np.uint32] = 100e-9
```

Comment:
I am not sure if there is any advantage of using `np.uint32` here, maybe just using Python's `int` would also work and is simpler.